split/groupresource.go

```diff
@@ -0,0 +1,160 @@
+package split
+
+import (
+	"fmt"
+	"io/fs"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/go-openapi/spec"
+	"github.com/iancoleman/strcase"
+
+	"github.com/kubewarden/k8s-objects-generator/object_templates"
+	"github.com/kubewarden/k8s-objects-generator/swagger_helpers"
+)
+
+const (
+	kubernetesGroupVersionKindKey = "x-kubernetes-group-version-kind"
+)
+
+type groupVersionResource struct {
+	Group   string
+	Version string
+	Kind    string
+}
+
+func GenerateGroupResources(project Project, plan *RefactoringPlan) error {
+	objectKindTemplate, err := template.New("objectKind").Parse(object_templates.ObjectKindTemplate)
+	if err != nil {
+		return err
+	}
+	groupVersionTemplate, err := template.New("groupVersion").Parse(object_templates.GroupVersionTemplate)
+	if err != nil {
+		return err
+	}
+
+	var lastGVK *groupVersionResource
+	var gvkCount int
+	for _, pkg := range plan.Packages {
+		log.Println("============================================================================")
+		log.Println("Generating GVK resources for module", pkg.Name)
+		for _, dfn := range pkg.Definitions {
+			gvk, err := groupKindResource(dfn)
+			if err != nil {
+				return err
+			}
+
+			if gvk != nil {
+				gvkCount++
+				objectKindFilePath := filepath.Join(project.Root, dfn.PackageName, fmt.Sprintf("%s_gvk.go", strcase.ToSnake(gvk.Kind)))
+				if err = generateResourceFile(objectKindFilePath, objectKindTemplate, gvk); err != nil {
+					return err
+				}
+				lastGVK = gvk
+			}
+		}
+		if lastGVK != nil {
+			groupVersionFilePath := filepath.Join(project.Root, pkg.Name, "group_version.go")
+			if err = generateResourceFile(groupVersionFilePath, groupVersionTemplate, lastGVK); err != nil {
+				return err
+			}
+
+			log.Printf("Generated GVK resources for module %s (visited %d/%d)", pkg.Name, gvkCount, len(pkg.Definitions))
+			log.Println("Generated GV resource for module", pkg.Name)
+			lastGVK, gvkCount = nil, 0
+		}
+	}
+
+	return copyStaticFiles(project.Root)
+}
+
+func generateResourceFile(path string, templ *template.Template, gvk *groupVersionResource) error {
+	gvkFile, err := os.OpenFile(path, os.O_CREATE|os.O_RDWR, 0644)
+	if err != nil {
+		return err
+	}
+	defer gvkFile.Close()
+
+	return templ.Execute(gvkFile, gvk)
```

Comment:
When something goes wrong, it would be nice to know which template caused the error